Set/Models/GameModel.swift

```diff
@@ -47,18 +47,40 @@ struct GameModel {
         playedCards = []
     }
     
-    mutating func checkASet(setOfCards: [CardModel]) {
+    private func checkASet(for setOfCards: [CardModel]) -> Bool {
         print("checkASet func is called")
-        for card in setOfCards {
-            print("checked")
-        }
+        return true
     }
-    
-    
-    mutating func toggle(cardId: UUID) {
-        guard let chousenIndex = cardsOnTheScreen.firstIndex(where: {$0.id == cardId}) else {
+
+    mutating func toggleCard(by cardId: UUID) {
+        guard let chousenIndex = cardsOnTheScreen.firstIndex(where: { $0.id == cardId }) else {
+            return
+        }
+        
+        cardsOnTheScreen[chousenIndex].isSelected.toggle()
+        let selectedCards = cardsOnTheScreen.filter { $0.isSelected == true }
+                
+        guard selectedCards.count >= 3 else {
             return
         }
-        cardsOnTheScreen[chousenIndex].isActive.toggle()
+        
+        guard selectedCards.count == 3 else {
+            for _ in 0 ..< selectedCards.count {
+                var cardToReset = cardsOnTheScreen.filter { $0.isSelected == true }.first
+                cardToReset?.isSelected.toggle()
+            }
+            return
+        }
+        
+        if checkASet(for: selectedCards) {
+            submitASet(for: selectedCards)
+        }
+    }
+    
+    private mutating func submitASet(for cards: [CardModel]) {
+        print("Make cards green")
```

Comment:
This is a good practice to leave a print in a developing process to have uderstanding what is going on